addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/ExternalRedirectScanRule.java

```diff
@@ -379,31 +420,23 @@ private int isRedirected(String payload, HttpMessage msg) {
         // (4) Check if redirection occurs by Base Tag
         // http://code.google.com/p/html5security/wiki/RedirectionMethods
         // <base href="http://evil.com/" />
-        //
 
         // (5) Check if redirection occurs by Javascript
         // http://code.google.com/p/html5security/wiki/RedirectionMethods
-        // location='http://evil.com/';
-        // location.href='http://evil.com/';
-        // location.reload('http://evil.com/');
-        // location.replace('http://evil.com/');
-        // location.assign('http://evil.com/');
-        // window.open('http://evil.com/');
-        // window.navigate('http://evil.com/');
-        //
         if (StringUtils.indexOfIgnoreCase(content, payload) != -1) {
             List<Element> jsElements = htmlSrc.getAllElements(HTMLElementName.SCRIPT);
-            String matchingUrl = "(\\Q" + payload + "\\E|\\Qhttp://" + REDIRECT_SITE + "\\E)";
             Pattern pattern;
 
             for (Element el : jsElements) {
                 value = el.getContent().toString();
 
                 // location='http://evil.com/';
                 // location.href='http://evil.com/';
-                pattern =
-                        Pattern.compile("(?i)location(\\.href)?\\s*=\\s*('|\")\\s*" + matchingUrl);
-                if (pattern.matcher(value).find()) {
+                pattern = Pattern.compile("(?i)location(?:\\.href)?\\s*=\\s*['\"](.*)['\"]");
+                Matcher jsAssignmentMatcher = pattern.matcher(value);
+                if (jsAssignmentMatcher.find()
+                        && StringUtils.startsWithIgnoreCase(
+                                jsAssignmentMatcher.group(1), HttpHeader.HTTP)) {
```

Comment:
I’ll have to read through all of these. I guess it could have been a check for HTTP or HTTPS but since one is a substring of the other it seemed fine. The main point per the User Group thread was that if there isn’t a scheme it’s treated as a relative URL.

Reply:
True that covers both schemes (doh!), the first part of the comment still applies though. Does the relative URL apply here too? These are not the same cases as mentioned in that thread.

Reply:
I believe I tested relative vs. absolute while putting this together, but I'll revisit.

Reply:
So yes the relative thing still applies here. If you try to `location.href = "https://www.w3schools.com";` that works fine, if you try `location.href = "www.w3schools.com";` then it'll likely 404 (depending where you're browser is sitting to start with).

Reply:
I have made a tweak to check the site.